### PR TITLE
[IMP] mail: added edit feature for message

### PR DESCRIPTION
**Pull request by pka-odoo**
- added edit icon in message core if the current user is author of the message
- added more button in message core, that contains edit button
- edit the message with emoji and attachment
- inline composer for desktop device and using existing composer for
  mobile devices

taskId: 2362167

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/components/message/message.scss [hunk 1]
```diff
@@ -77,6 +78,7 @@
 .o_Message_headerCommands {
     margin-inline-end: map-get($spacers, 2);
     align-self: center;
+    border: none !important;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/message/message.scss:81]: important is a keyword we should never use, if this has to override something, it's better to "remove the something" instead or add it only conditionally

File: addons/mail/static/src/components/message/message.xml [hunk 2]
```diff
@@ -38,19 +38,6 @@
...
-                                    t-att-class="{
-                                        'fa-star': message.isStarred,
-                                        'fa-star-o': !message.isStarred,
-                                        'o-message-selected': isSelected,
-                                        'o-message-starred': message.isStarred,
-                                        'o-mobile': env.messaging.device.isMobile,
-                                    }" t-on-click="_onClickStar"
-                                />
-                            </t>
-                        </div>
```
Comment (seb-odoo) [inline addons/mail/static/src/components/message/message.xml:53]: why is all of this moved?

File: addons/mail/static/src/models/composer/composer.js [hunk 3]
```diff
@@ -367,6 +372,63 @@ function factory(dependencies) {
...
+
+            const data = this.env.models['mail.message'].convertData(messageData);
+            this.message.update(Object.assign(data, { isEditingMessage: false }));
+            if (this.thread) {
+                this.update({
+                    textInputContent: "",
+                    isLastStateChangeProgrammatic: true,
+                });
+            }
+        }
```
Comment (seb-odoo) [inline addons/mail/static/src/models/composer/composer.js:430]: I don't like that we need to handle this manually. I still think we should postpone this task until we have the new web editor ready to use.

**Review by seb-odoo — CHANGES_REQUESTED** [review]
Hello, here is a quick check of the code as you requested it. I gave some very specific comments on some lines, but overall I think we are not ready technically to make this task.

The main issue is that once the message exits, its content is html, and the only proper way to allow editing html is to have the web editor. And we are not allowed to use the existing one, because we know we will replace it as soon as possible. (There is nothing to do but wait.)

The second issue (which can be fixed from us) is that there is currently a problem with the composer model and the message model. I see what you have been forced to do to for this PR and it is not easy to maintain.
First we should have one composer model per composer component, instead of one composer model per thread model.
Second composer should always be modeling a message, and it should be able to distinguish from a new message or an existing message more easily, and with a lot more common code between the two.

**Comment by tde-banana-odoo** [comment]
Done (with a nice commit message) at https://github.com/odoo/odoo/pull/76483

**Pull request closed without merging**